lib/rules.js

```diff
@@ -184,6 +227,185 @@ export const MULTIPLY_BY_INVERSE = defineRuleString('#a / (#b / #c)', '#a * (#c
 export const ABSOLUTE_VALUE = defineRuleString('|-#a|', '#a')
 
+// ROOT
+
+// assumes that the exponent is a constant value
+// TODO: handle cases like nthRoot('x^2y', 2)
+// e.g. nthRoot(x^2, 4) -> nthRoot(x^1 ,2)
+export const CANCEL_EXPONENT = defineRule(
+    (node) => {
+        let isNthRoot = false
+        let validRadicand = false
+        if (query.isApply(node)) {
+            isNthRoot = (node.op == 'nthRoot')
+            validRadicand = query.isPow(node.args[0])
+        }
+        return (isNthRoot && validRadicand) ? {node} : null
+    },
+    (node) => {
+        const radicand = node.args[0]
+        const variable = radicand.args[0]
+        const exponent = radicand.args[1]
+        let index = node.args[1]
+
+        // simplify exponent / index
+        // e.g. nthRoot(x^2, 4) -> 2/4 -> 1/2
+        const newRoot = applyRule(SIMPLIFY_FRACTION,
+                                  build.applyNode('div', [exponent, index]))
+
+        let numerator = newRoot.args[0]
+        let denominator = newRoot.args[1]
+        let numVal = query.getValue(numerator)
+        let denomVal = query.getValue(denominator)
+
+        // Case #1: numerator > denominator || numerator == -1 && denominator = 1
+        // e.g nthRoot(x^4, 2) -> 4/2 -> 2/1 -> x^2
+        // Case #2: numerator == denominator
+        // e.g nthRoot(x^2, 2) -> 2/2 -> x^1
+        // Case #3: numerator < denominator ||
+        // numerator > denominator && denominator != 1, return nthRoot
+        // e.g nthRoot(x^2, 4) -> 2/4 -> 1/2 -> nthRoot(x^1, 2)
+        if ((numVal > denomVal || numVal == -1) && denomVal == 1) {
+            return build.applyNode('pow', [variable, numerator])
+        } else if (numVal === denomVal) {
+            return build.applyNode('pow', [variable, build.numberNode(1)])
+        } else {
+            return build.applyNode (
+                'nthRoot',
+                [build.applyNode('pow', [variable, numerator]), denominator]
+            )
+        }
+    }
+)
+
+// e.g. nthRoot(2, 2) * nthRoot(3, 2) -> nthRoot(2 * 3, 2)
+export const COMBINE_UNDER_ROOT = defineRuleString('nthRoot(#a_0, #b) * ...', 'nthRoot(#a_0 * ..., #b)')
+
+// e.g. 2^1 * 2^1 * 2^3 -> 2 ^ 3
+export const CONVERT_MULTIPLICATION_TO_EXPONENT = defineRuleString('#a^#b_0 * ...', '#a^#eval(#b_0 + ...)', {a: query.isNumber, b: query.isNumber})
+
+// e.g. nthRoot(2 * x, 2) -> nthRoot(2, 2) * nthRoot(x, 2)
+export const DISTRIBUTE_NTH_ROOT = defineRuleString('nthRoot(#a_0 * ..., #b)', 'nthRoot(#a_0, #b) * ...')
+
+
+// TODO: #10 and #23 (`or` rules and applying multiple rules if
+// multiple occurrences)
+// e.g. nthRoot(4, 2) * nthRoot(x^2, 2) -> 2 * x
+export const EVALUATE_DISTRIBUTED_NTH_ROOT = defineRule(
+    (node) => {
+        let isDistributed = query.isMul(node)
+        let canEvaluate = false
+        if (isDistributed) {
+            canEvaluate = node.args.every(arg => canApplyRule(CANCEL_EXPONENT, arg) || canApplyRule(NTH_ROOT_VALUE, arg))
+        }
```

Comment:
This will fail on things like `x * nthRoot(4, 2) * nthRoot(x^2, 2) * y`.  Please add a test case for this but comment it out.  We can enable it once we have #10.  It would be nice if there was a way to apply a rule from within a pattern.  I've opened an issue for this https://github.com/semantic-math/math-rules/issues/39.